lib/tasks/knows_what_exercises_are_called_and_where_they_live_in_the_directory_structure.rb

```diff
@@ -0,0 +1,34 @@
+class Exercise
+  def self.all
+    FileList['exercises/*'].pathmap('%f').map { |e| new(e) }
```

Comment:
I like that it returns instances, and how  you've used `to_s` to make it do the sensible thing.

I would extract an `exercise_names` method.